src/network/clientpackethandler.cpp

```diff
@@ -1803,20 +1803,32 @@ void Client::handleCommand_SetLighting(NetworkPacket *pkt)
 {
 	Lighting& lighting = m_env.getLocalPlayer()->getLighting();
 
-	if (pkt->getRemainingBytes() >= 4)
-		*pkt >> lighting.shadow_intensity;
-	if (pkt->getRemainingBytes() >= 4)
-		*pkt >> lighting.saturation;
-	if (pkt->getRemainingBytes() >= 24) {
-		*pkt >> lighting.exposure.luminance_min
-				>> lighting.exposure.luminance_max
-				>> lighting.exposure.exposure_correction
-				>> lighting.exposure.speed_dark_bright
-				>> lighting.exposure.speed_bright_dark
-				>> lighting.exposure.center_weight_power;
-	}
-	if (pkt->getRemainingBytes() >= 4)
-		*pkt >> lighting.volumetric_light_strength;
-	if (pkt->getRemainingBytes() >= 4)
-		*pkt >> lighting.shadow_tint;
+	if (pkt->getRemainingBytes() < 4)
```

Comment:
Hmm, I looked into the packet structure in `networkpacket.h/cpp` and I've got doubts if there is a real necessity to check the remaining bytes while reading from the package in this function at all (it relates also to all lighting parameters read here). As I see in `Server::SendSetLighting` the lighting structure is sent *always* entirely even if only one parameter was changed. Therefore, the package received by the client will have always the same count of bytes independently from which parameters the modder passed in the table and it turns out checking for which of them were actually packed is redudant?

Reply:
This is for backwards compatibility, because old servers send less data.